db/version_set.cc

```diff
@@ -3187,11 +3187,21 @@ void SortFileByOverlappingRatio(
                                           ttl_boost_score;
   }
 
-  std::sort(temp->begin(), temp->end(),
-            [&](const Fsize& f1, const Fsize& f2) -> bool {
-              return file_to_order[f1.file->fd.GetNumber()] <
-                     file_to_order[f2.file->fd.GetNumber()];
-            });
+  // We are going to find the files with highest score. Since we are going
+  // to pick files fro highest score to lower, it's unlikely that we would
+  // need to pick many files, so we limit files for this partial order.
+  // In case we use up all the sorted files, we are essentially pick files
+  // in random order, but it should be rare.
+  const size_t kTotalSortedElements = 8;
```

Comment:
For `kByCompensatedSize`, it sorts the top n (50) files:
https://github.com/facebook/rocksdb/blob/48ce44240c398850a752d13d1a27c9849ae6fad2/db/version_set.cc#L3229-L3233

can we reuse that and apply it to both `kByCompensatedSize` and `kMinOverlappingRatio`, or even to all the other strategies.